homeassistant/components/huawei_lte/select.py

```diff
@@ -0,0 +1,115 @@
+"""Support for Huawei LTE selects."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from functools import partial
+import logging
+
+from huawei_lte_api.enums.net import LTEBandEnum, NetworkBandEnum
+
+from homeassistant.components.select import (
+    DOMAIN as SELECT_DOMAIN,
+    SelectEntity,
+    SelectEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HuaweiLteBaseEntityWithDevice
+from .const import DOMAIN, KEY_NET_NET_MODE, NETWORKMODE_TO_STRING
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up from config entry."""
+    router = hass.data[DOMAIN].routers[config_entry.unique_id]
```

Comment:
Changed in 4c31317c0645f74d7d4b6a2f09e049258e2f8e46

```suggestion
    router = hass.data[DOMAIN].routers[config_entry.entry_id]
```

Reply:
Good catch, that's why my tests were not working. I was trying to do something similar to https://github.com/home-assistant/core/blob/dev/tests/components/plugwise/test_select.py and https://github.com/home-assistant/core/blob/dev/tests/components/plugwise/conftest.py to make the tests cover the whole initialization procedure, but I'm stuck in a situation where there are no services registered...

What do you think, does it make sense to move towards using fixtures (instead of manual patching as done in `test_switches`)?

Reply:
In principle, would very much like to head towards fixture based approach. Much appreciated if you're willing to do the ground work for that here :+1: